services/datamanager/datasync/sync.go

```diff
@@ -241,8 +242,7 @@ func exponentialRetry(cancelCtx context.Context, fn func(cancelCtx context.Conte
 		return nil
 	}
 	// Don't retry non-retryable errors.
-	s := status.Convert(err)
-	if s.Code() == codes.InvalidArgument {
+	if !isRetryableGRPCError(err) {
```

Comment:
Basically we were missing the check and would keep retrying to upload the bad file due to our retry logic, but checking against `proto.Error` does in fact work!